app/controllers/questionnaire_controller.rb

```diff
@@ -158,15 +187,17 @@ def set_is_mentor
   end
 
   def check_response(response)
-    render(status: 404, plain: 'De vragenlijst kon niet gevonden worden.') && return unless response
-    render(status: 404, plain: 'Je hebt deze vragenlijst al ingevuld.') && return if response.completed_at
-    render(status: 404, plain: 'Deze vragenlijst kan niet meer ingevuld worden.') if response.expired?
-  end
+    unless response
+      render(status: 404, plain: 'De vragenlijst kon niet gevonden worden.')
+      return
+    end
 
-  def check_invitation_token(invitation_token)
-    render(status: 404, plain: 'De vragenlijst kon niet gevonden worden.') && return unless invitation_token
-    render(status: 404, plain: 'Je hebt deze vragenlijst al ingevuld.') && return if
-      invitation_token.response.completed_at
-    render(status: 404, plain: 'Deze vragenlijst kan niet meer ingevuld worden.') if invitation_token.response.expired?
+    # Instead of throwing a 404, just redirect to the next page in line if one is already completed.
+    if response.completed_at
+      redirect_to_next_page
+      return
+    end
+
+    render(status: 404, plain: 'Deze vragenlijst kan niet meer ingevuld worden.') if response.expired?
```

Comment:
Ik zou hier ook gewoon zeggen: redirect_to_next_page, eventueel met een flash[:error] = 'Deze vragenlijst kan niet meer ingevuld worden'

Als je de `my_open_responses` functie aanpast, dan zou je hier in principe namelijk alleen kunnen komen als de response expiret in de tijd van de redirect, of, (tijdens de create) bij het invullen van een vragenlijst als de response geexpired is in de tijd dat men de vragenlijst invulde. In beide gevallen lijkt het mij wel cool om dan gewoon door te gaan naar de volgende vragenlijst (als die er is) of de rewardspagina/mentordashboard, ipv mensen een "dead end" te geven.